+nosnoc/+discrete_problem/stewart.m

```diff
@@ -0,0 +1,771 @@
+                            Hij = {};
+                            for kk=1:opts.n_s
+                                theta_ijk = obj.w.theta(ii,jj,kk);
+
+                                Gij = vertcat(Gij, {sum_lambda});
+                                Hij = vertcat(Hij, {theta_ijk});
+                            end
+                            obj.G.cross_comp(ii,jj) = {vertcat(Gij{:})};
+                            obj.H.cross_comp(ii,jj) = {vertcat(Hij{:})};
+                            lambda_prev = obj.w.lambda(ii,jj,opts.n_s);
+                        end
+                    end
+                  case CrossCompMode.STAGE_FE
+                    lambda_prev = obj.w.lambda(0,0,opts.n_s);
+                    % Do cross comp for distance with lambda
+                    for ii=1:opts.N_stages
+                        for jj=1:opts.N_finite_elements(ii);
+                            sum_theta = sum2(obj.w.theta(ii,jj,:));
+                            Gij = {lambda_prev};
+                            Hij = {sum_theta};
+                            for kk=1:(opts.n_s + rbp)
+                                lambda_ijk = obj.w.lambda(ii,jj,kk);
+
+                                Gij = vertcat(Gij, {lambda_ijk});
+                                Hij = vertcat(Hij, {sum_theta});
+                            end
+                            obj.G.cross_comp(ii,jj) = {vertcat(Gij{:})};
+                            obj.H.cross_comp(ii,jj) = {vertcat(Hij{:})};
+                            lambda_prev = obj.w.lambda(ii,jj,opts.n_s);
+                        end
+                    end
+                  case CrossCompMode.FE_FE
+                    lambda_prev = obj.w.lambda(0,0,opts.n_s);
+                    % Do cross comp for distance with lambda
+                    for ii=1:opts.N_stages
+                        for jj=1:opts.N_finite_elements(ii);
+                            sum_lambda = lambda_prev + sum2(obj.w.lambda(ii,jj,:));
+                            sum_theta = sum2(obj.w.theta(ii,jj,:));
+                            obj.G.cross_comp(ii,jj) = {sum_lambda};
+                            obj.H.cross_comp(ii,jj) = {sum_theta};
+                            lambda_prev = obj.w.lambda(ii,jj,opts.n_s);
+                        end
+                    end
+                end
+            else
+                for ii=1:opts.N_stages
+                    for jj=1:opts.N_finite_elements(ii);
+                        Gij = {};
+                        Hij = {};
+                        for kk=1:opts.n_s
+                            lambda_ijk = obj.w.theta(ii,jj,kk);
+                            theta_ijk = obj.w.theta(ii,jj,kk);
+
+                            obj.G.standard_comp(ii,jj, kk) = {lambda_ijk};
+                            obj.H.standard_comp(ii,jj, kk) = {theta_ijk};
+                        end
+                    end
+                end
+            end
+        end
+
+        function step_equilibration(obj)
```

Comment:
rename,e.g. generate_step_equilibration_constraints (also rename the two above functions in the same style)